### PR TITLE
[TT-11124, TT-11070]: update tyk-pump with graph pumps

### PR DESCRIPTION
**Pull request by kofoworola**
## Description

This PR adds a few changes:
- Adds the Graph SQL pump if `postgres-pump` or `postgres` is added to the list of  `pump.backend` in the values
- Adds the SQL graph aggregate pump if `postgres` or `postgres-aggregate` is in the list of `pump.backend`
- Adds the Mongo graph pump if `mongo` is in the list of `pump.backend`

### graph SQL pumps:
- Install tyk-stack with [postgresql](https://github.com/TykTechnologies/tyk-charts/tree/main/tyk-stack#quick-start-with-postgresql)
-  Modify pump, gateway and dashboard image tags manually in cluster to latest release candidates which are v1.9.0-rc1, v5.3alpha11 and v5.3alpha11 respectively
- Create a graphql API on the dashboard that points to trevorblades and run a sample request e.g 
```graphql
{
  countries{
    code
  }
}
```
- Check activity by graph page on the dashboard it should be populated. THis is a sign the SQL graph aggregate pump works
- To confirm the SQL graph pump, ssh into your postgres instance and check if the table `tyk_analytics_graphs` has been populated with information regarding the earlier request

### Mongo Pumps
- Install tykstack with MongoDB as the backend
- Modify pump, gateway and dashboard image tags manually in cluster to latest release candidates which are v1.9.0-rc1, v5.3alpha11 and v5.3alpha11 respectively
- Create a graphql API on the dashboard that points to trevorblades and run a sample request e.g 
```graphql
{
  countries{
    code
  }
}
```
- check the mongo db for the collection `tyk_graph_analytics` unless otherwise specified, there should be populated values in there

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: components/tyk-pump/templates/deployment-pmp.yaml [hunk 1]
```diff
@@ -131,6 +131,19 @@ spec:
               secretKeyRef:
                 name: {{ include "tyk-pump.pg_connection_string_secret_name" . }}
                 key: {{ include "tyk-pump.pg_connection_string_secret_key" . }}
+
+          #SQL Graph Pump
+          - name: TYK_PMP_PUMPS_SQLGRAPH_TYPE
+            value: "sql-graph"
+          - name: TYK_PMP_PUMPS_SQLGRAPH_META_TYPE
+            value: "postgres"
+          - name: TYK_PMP_PUMPS_SQLGRAPH_META_TABLENAME
```
Comment (komalsukhani) [inline components/tyk-pump/templates/deployment-pmp.yaml:140]: @kofoworola Should this be configurable?

**Comment by komalsukhani** [comment]
@kofoworola Can you add in the description how you tested it?

File: components/tyk-pump/templates/deployment-pmp.yaml (hunk 1 above)
Comment (kofoworola) [reply components/tyk-pump/templates/deployment-pmp.yaml:140]: We don't need it to be, but i can add a value for it and assign a default value

File: components/tyk-pump/templates/deployment-pmp.yaml [hunk 2]
```diff
@@ -157,6 +181,21 @@ spec:
                 key: {{ include "tyk-pump.mongo_url_secret_key" . }}
           - name: TYK_PMP_PUMPS_MONGO_META_MONGODRIVERTYPE
             value: "{{ default "mgo" .Values.global.mongo.driver}}"
+
+          # Mongo Graph Pump
+          - name: TYK_PMP_PUMPS_GRAPHMONGO_TYPE
+            value: "mongo-graph"
+          - name: TYK_PMP_PUMPS_GRAPHMONGO_META_COLLECTIONNAME
+            value: "tyk_graph_analytics_pump"
```
Comment (komalsukhani) [inline components/tyk-pump/templates/deployment-pmp.yaml:189]: @kofoworola We will need to make this configurable too.

Comment (kofoworola) [reply components/tyk-pump/templates/deployment-pmp.yaml:189]: okay, will add that change

File: components/tyk-pump/values.yaml [hunk 3]
```diff
@@ -244,6 +244,9 @@ pump:
       podMonitorSelector:
         release: prometheus-stack
 
+  sqlGraphPump:
```
Comment (komalsukhani) [inline components/tyk-pump/values.yaml:247]: Can you add description for this field?

**Comment by komalsukhani** [comment]
@kofoworola Can you also update the readmes of tyk-pump and tyk-stack chart?
It will also need a change on [tyk-docs](https://tyk.io/docs/deployment-and-operations/tyk-self-managed/deployment-lifecycle/installations/kubernetes/tyk-helm-tyk-stack/) too.

**Review by komalsukhani — APPROVED** [review]

**Pull request merged**